fsharp-backend/src/LibExecution/StdLib/LibList.fs

```diff
@@ -962,123 +963,97 @@ let fns : List<BuiltInFn> =
-//                         let err_details =
-//                           match v with
-//                           | DList l ->
-//                               Printf.sprintf
-//                                 "It has length %i but must have length 2."
-//                                 (List.length l)
-//                           | non_list ->
-//                               let tipe =
-//                                 non_list
-//                                 |> Dval.tipe_of
-//                                 |> Dval.tipe_to_developer_repr_v0
-//                               in
-//                               Printf.sprintf
-//                                 "It is of type `%s` instead of `List`."
-//                                 tipe
-//                         in
-//                         Error
-//                           (DError
-//                              ( SourceNone
-//                              , Printf.sprintf
-//                                  "Expected every value within the `pairs` argument passed to `%s` to be a list with exactly two values. However, that is not the case for the value at index %i: %s. %s"
-//                                  state.executing_fnname
-//                                  (idx_from_rev_idx rev_idx l)
-//                                  (Dval.to_developer_repr_v0 v)
-//                                  err_details )))
-//               in
-//               let result =
-//                 (* We reverse here so that the [foldi] consing happens in the correct order.
-//                 * It does mean that the index passed by [foldi] counts from the end *)
-//                 l |> List.rev |> List.foldi (Ok ([], [])) fold_fn
-//               in
-//               ( match result with
-//               | Ok (res_a, res_b) ->
-//                   DList [DList res_a; DList res_b]
-//               | Error v ->
-//                   v )
-//           | _ ->
-//               incorrectArgs ())
-//     ; sqlSpec = NotYetImplementedTODO
-//     ; previewable = Pure
-//     ; deprecated = NotDeprecated }
+    { name = fn "List" "zipShortest" 0
+      parameters = [ Param.make "as" (TList varA) "";  Param.make "bs" (TList varB) "" ]
+      returnType = TList varA
+      description = "Returns a list of parallel pairs from `as` and `bs`.
+        If the lists differ in length, values from the longer list are dropped.
+        For example, if `as` is `[1,2]` and `bs` is `[\"x\",\"y\",\"z\"]`, returns `[[1,\"x\"], [2,\"y\"]]`.
+        Use `List::zip` if you want to enforce equivalent lengths for `as` and `bs`.
+        See `List::unzip` if you want to deconstruct the result into `as` and `bs` again."
+      fn =
+        (function
+        | state, [ DList l1; DList l2 ] ->
+            // We have to do this munging because OCaml's map2
+            // and Fsharp's zip enforces lists of the same length
+            let len = min (bigint (List.length l1)) (bigint (List.length l2))
```

Comment:
I thought that it was needed because the functions `List::getAt`, `List::drop`, and `List::take` were using that cast for `List.length`. But I saw now that it isn't necessary in all cases for `List.length`.